src/configuration/decoration.ts

```diff
@@ -102,6 +130,18 @@ class DecorationImpl {
       borderWidth: '1px',
     });
 
+    this.marks = this._markNames.reduce(
+      (marks, name) => ({
+        ...marks,
+        [name]: vscode.window.createTextEditorDecorationType({
```

Comment:
Probably it makes more sense to cache the `TextEditorDecorationType`s. IIRC, these take a non-negligible time to construct if you're doing it often.